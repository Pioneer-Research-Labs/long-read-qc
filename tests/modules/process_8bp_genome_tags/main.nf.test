nextflow_process {

    name "Test Process process_8bp_genome_tags"
    script "modules/process_8bp_genome_tags/main.nf"
    process "process_8bp_genome_tags"

    test("Should generate several fastq files by genome") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
               input[0] = Channel.fromList([ tuple( [id:'test_sample'],
                file('${projectDir}/test_data/test_extracted_genome_tags.tsv'),
                file('${projectDir}/test_data/Tesseract_Barcode_Tracking_Grid.csv'),
                file('s3://pioneer-scratch/dummy/dummy_seqs/test.fastq.gz'),
                file('${projectDir}/test_data/c.00432.dna') )])
                """
            }
        }

        then {
            assert process.success
            assert path("$outputDir/test_sample/test_sample_sample_sheet.csv").readLines().size() == 11
            assert snapshot("$outputDir/test_sample/test_sample_sample_sheet.csv").match()
        }

    }

}
