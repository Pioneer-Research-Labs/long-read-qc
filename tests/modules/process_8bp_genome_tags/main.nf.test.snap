{
    "Should generate several fastq files by genome": {
        "content": [
            "test_sample_sample_sheet.csv:md5,3af29ebe2332a427d2794ff7f1a7d53e"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-22T12:45:45.158171"
    }
}